tests/phpunit/tests/test-canonical-domain.php

```diff
@@ -14,23 +14,29 @@ public static function wpSetUpBeforeClass( WP_UnitTest_Factory $factory ) {
 	protected function init( $domain ) {
 		global $wp_rewrite;
 
-		$domains = array(
-			'en' => 'http://example.org',
-			'fr' => $domain,
+		$options = array_merge(
+			self::$model->options->get_all(),
+			array(
+				'hide_default' => true,
+				'force_lang'   => 3,
+				'domains'      => array(
+					'en' => 'http://example.org',
+					'fr' => $domain,
+				),
+			)
 		);
-
-		self::$model->options['hide_default'] = 1;
-		self::$model->options['force_lang'] = 3;
-		self::$model->options['domains'] = $domains;
+		update_option( 'polylang', $options );
```

Comment:
Looks tricky, could `self::create_options();` be made *before* `array_merge()`?

Reply:
No, that's the whole point. The point is to have the domains set in the options BEFORE calling `self::create_options()` to avoid having to merge them, and so, calling the sanitization process, which requires `$GLOBALS['polylang']` and the languages to be ready (`_doing_it_wrong()`).

Reply:
Improved in [37ee8c9](https://github.com/polylang/polylang/pull/1481/commits/37ee8c9e8a61d2efd62e24dd25c76c47e1f3333c).